### ISSUE TITLE
Rework video/audio timestamping

### ISSUE THREAD

**Issue by ghost** [bluecherrydvr/bluecherry-apps-issues#55]

**Comment by ghost** [comment]
Currently the system sometimes produces files that don't play correctly.

**Comment by antonsviridenko** [comment]
172.16.1.132.ACTi ACM-8211.2014-10-18 02-46-30.mkv

as an example

Duration displayed in bluecherry client events is 2:29, when downloaded it displays 1 second duration in Totem, and 0:00 in vlc and smplayer

**Comment by antonsviridenko** [comment]
some piece from server debug logs, can be related to this issue

```
Oct 17 21:11:32 1204-amd64 bc-server[29620]: D(3/ACTi ACM-8211): [libav] debug Writing block at offset 53754031, size 10940, pts 426960, dts 426960, duration 125, flags 0
Oct 17 21:11:32 1204-amd64 bc-server[29620]: D(3/ACTi ACM-8211): [libav] debug Writing block at offset 53764978, size 10484, pts 427050, dts 427050, duration 125, flags 0
Oct 17 21:11:32 1204-amd64 bc-server[29620]: D(3/ACTi ACM-8211): [libav] error Application provided invalid, non monotonically increasing dts to muxer in stream 0: 1092758400 >= 1092758400
```

**Comment by ghost** [comment]
WIP in branch "ts_fix", has positive results, needs cleanup before merging.

**Comment by ghost** [comment]
Merged in to master as commits 8f72a8564b911f6dbc149955130ff6248eaad60a, 70f08a4a29dfe883b4c6b545de45eec468bd870c, 1d2491e52fe1a8833658ec9f2df6bf7d6327dd6e, 949e9774473d946b8ec27818de6a34a1bcd5b4d4
